pkg/storage/interfaces.go

```diff
@@ -167,4 +167,6 @@ type Interface interface {
 	//    }
 	// })
 	GuaranteedUpdate(ctx context.Context, key string, ptrToType runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc) error
+
+	Xxx(ctx context.Context, key string, out runtime.Object, ignoreNotFound bool, precondtions *Preconditions, tryUpdate UpdateFunc, guess runtime.Object) error
```

Comment:
This actually encompass GUpdate().
It's an optimizaiton for GUpdate().
We should modify GUpdate() instead, even though GUpdate() is a ridiculously complicated interface already.